ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/Utils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2015 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3cli.util;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class Utils {
+
+    public static ImmutableList<Path> listObjectsForDirectory(final Path directory) throws IOException {
+        final List<Path> objects = new ArrayList<>();
```

Comment:
This should be an `ImmutableList.Builder<Path>`  There are several examples of doing this here: https://github.com/SpectraLogic/ds3_java_sdk/blob/master/ds3-sdk/src/main/java/com/spectralogic/ds3client/utils/ResponseUtils.java#L22